fleet/agent_group_service.go

```diff
@@ -21,8 +21,22 @@ var (
 	ErrMaintainAgentGroupChannels = errors.New("failed to maintain agent group channels")
 )
 
-func (svc fleetService) addAgentsToAgentGroupChannel(token string, g AgentGroup) error {
+func (svc fleetService) removeAgentGroupSubscriptions(groupID string, ownerID string) error {
+	list, err := svc.agentRepo.RetrieveAllByAgentGroupID(context.Background(), ownerID, groupID, true)
```

Comment:
see comment in RemoveAgentGroup. in addition, even though for new group creation we have to notify all of the agents individually, for group removal we do not because all of those asians should already be listening on the group channel. so instead of notifying each agent individually on the agent channel, we can send a single notification on the group channel.